frontend/src/LandingPage.js

```diff
@@ -1,50 +1,269 @@
+                initMap();
+            })
+        } else {
+            initMap();
+        }
+    });
+    function removeMarker(location) {
+        // Convert string location to JSON object.
+        const position = {
+            lat: parseFloat(location.split(',')[0]),
+            lng: parseFloat(location.split(',')[1]),
+        };
+
+        // Before updating markers clear all markers.
+        for (let i = 0; i < markers.length; i++) {
+          markers[i].markerObject.setMap(null);
+        }
+
+        // Find the specific marker and remove from the array.
+        for (let i = 0; i < markers.length; i++) {
+          if (markers[i].position.lat == position.lat &&
+              markers[i].position.lng == position.lng) {
+            markers.splice(i,1);
+            break;
+          }
+        }
+
+        // Update the markers on map page.
+        for (let i = 0; i < markers.length; i++) {
+          markers[i].markerObject.setMap(map);
+        }
+    }
+    function addMarker(location) {
+        // Convert string location to JSON object.   
+        const position = {
+            lat: parseFloat(location.split(',')[0]),
+            lng: parseFloat(location.split(',')[1]),
+        };
+        map.setCenter(position);
+
+        const markerObject = new google.maps.Marker({
+          position: position,
+        });
+
+        const marker = {
+          position: position,
+          markerObject: markerObject,
+        }
+
+        // Add new marker to the array.
+        markers.push(marker);
+
+        // Update markers on map page.
+        for (let i = 0; i < markers.length; i++) {
+          markers[i].markerObject.setMap(map);
+        }
+    }
+    let lat1, lat2, long1, long2;
+    const [hiddenIn, setHiddenIn] = useState(false);
+    const [hiddenOut, setHiddenOut] = useState(true);
+    function authentication() {
+        
+        fetch('/api/login')
+            .then((response) => response.json())
+            .then((user) => {
+                const loginSection = document.getElementById('loginsection');
+                const loginMessage = document.getElementById('login');
+                const logoutSection = document.getElementById('logoutsection');
+                const logoutMessage = document.getElementById('logout');
+                
+                if (user.loggedIn) {
+                    loginSection.style.display = 'none';
+                    logoutMessage.href = user.url;
+                    logoutSection.style.display = 'block';
+                    
+                } else {
+                    logoutSection.style.display = 'none';
+                    loginMessage.href = user.url;
+                    loginSection.style.display = 'block';
+                    
+                }
+                currentUserId = user.id;
+                localStorage.setItem("id", currentUserId);
+            });
+    }
+
+    function calls() {
+        handleScriptLoad1(setQuery1);
```

Comment:
Usually it's a bad practice to use the name like smth1 smth2. It's so easy to make a typo here. If they are logically first and second, you can use _first and _second endings. But seems like they can have more clear names, like start place and end place.

Reply:
Fixed, thanks.

Reply:
Sorry about that, I called them like that so that I first see if it works and I pushed it like this. I struggled with making it work for both autocomplete input fields and that s why I named them one and two.